src/github/github.installation.controller.ts

```diff
@@ -0,0 +1,203 @@
+    private readonly githubInstallationService: GithubInstallationService,
+  ) {}
+
+  @Post()
+  async create(
+    @Body() createOrgDto: CreateOrgDto,
+  ): Promise<AllowedOrganization> {
+    try {
+      return this.githubInstallationService.create(createOrgDto);
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
+    }
+  }
+
+  @Get()
+  async findAll(): Promise<AllowedOrganization[]> {
+    try {
+      return this.githubInstallationService.findAll();
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
+    }
+  }
+
+  @Get(':id')
+  async findOne(@Param('id') id: string): Promise<AllowedOrganization> {
+    try {
+      return this.githubInstallationService.findOne(id);
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
+    }
+  }
+
+  @Get('by-name/:organizationName')
+  async findByName(
+    @Param('organizationName') organizationName: string,
+  ): Promise<AllowedOrganization> {
+    try {
+      return this.githubInstallationService.findByName(organizationName);
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
+    }
+  }
+
+  @Put(':id')
+  async update(
+    @Param('id') id: string,
+    @Body() updateOrgDto: CreateOrgDto,
+  ): Promise<AllowedOrganization> {
+    try {
+      return this.githubInstallationService.update(id, updateOrgDto);
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
+    }
+  }
+
+  @Delete(':id')
+  async remove(@Param('id') id: string): Promise<void> {
+    try {
+      return this.githubInstallationService.remove(id);
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
+    }
+  }
+
+  @Get('is-allowed/:organizationName')
+  async isOrganizationAllowed(
+    @Param('organizationName') organizationName: string,
+  ): Promise<boolean> {
+    try {
+      return this.githubInstallationService.isOrganizationAllowed(
+        organizationName,
+      );
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
```

Comment:
PullProofAI

Consider consolidating error handling logic to avoid repetition.
```diff
-       if (error.status) {
-         throw new HttpException(error.message, error.status);
-       } else {
-         throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
-       }
+       this.handleError(error);
```



<!-- This is an auto-generated comment by PullProofAI -->